src/pages/settings/Payments/AddDebitCardPage.js

```diff
@@ -0,0 +1,238 @@
+import React, {Component} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import {
+    View,
+    ScrollView,
+} from 'react-native';
+import HeaderWithCloseButton from '../../../components/HeaderWithCloseButton';
+import Navigation from '../../../libs/Navigation/Navigation';
+import ScreenWrapper from '../../../components/ScreenWrapper';
+import TextInputWithLabel from '../../../components/TextInputWithLabel';
+import styles from '../../../styles/styles';
+import StatePicker from '../../../components/StatePicker';
+import Text from '../../../components/Text';
+import TextLink from '../../../components/TextLink';
+import withLocalize, {withLocalizePropTypes} from '../../../components/withLocalize';
+import compose from '../../../libs/compose';
+import {addBillingCard} from '../../../libs/actions/PaymentMethods';
+import Button from '../../../components/Button';
+import KeyboardAvoidingView from '../../../components/KeyboardAvoidingView';
+import FixedFooter from '../../../components/FixedFooter';
+import Growl from '../../../libs/Growl';
+import {
+    isValidAddress, isValidExpirationDate, isValidZipCode, isValidDebitCard,
+} from '../../../libs/ValidationUtils';
+import CheckboxWithLabel from '../../../components/CheckboxWithLabel';
+
+const propTypes = {
+    /* Onyx Props */
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+};
+
+class DebitCardPage extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            nameOnCard: '',
+            cardNumber: '',
+            expirationDate: '',
+            securityCode: '',
+            billingAddress: '',
+            city: '',
+            selectedState: '',
+            zipCode: '',
+            acceptedTerms: false,
+        };
+
+        this.toggleTermsOfService = this.toggleTermsOfService.bind(this);
+        this.handleExpirationInput = this.handleExpirationInput.bind(this);
+        this.submit = this.submit.bind(this);
+    }
+
+    /**
+     * @returns {Boolean}
+     */
+    validate() {
+        if (this.state.nameOnCard === '') {
+            Growl.error(this.props.translate('addDebitCardPage.error.invalidName'));
+            return false;
+        }
+
+        if (!isValidDebitCard(this.state.cardNumber.replace(/ /g, ''))) {
+            Growl.error(this.props.translate('addDebitCardPage.error.debitCardNumber'));
+            return false;
+        }
+
+        if (!isValidExpirationDate(this.state.expirationDate)) {
+            Growl.error(this.props.translate('addDebitCardPage.error.expirationDate'));
+            return false;
+        }
+
+        if (!/^[0-9]{3,4}$/.test(this.state.securityCode)) {
```

Comment:
any reason we're not adding this to the constant file so it could be reused and then creating a method in the utlis?

Reply:
That's a good idea! Thanks, doing it now.